libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -40,11 +40,8 @@ static BOOL isNan(NSString *sourceString) {
 
 @interface NBPhoneNumberUtil ()
 
-@property(nonatomic, strong) NSLock *entireStringCacheLock;
-@property(nonatomic, strong) NSMutableDictionary *entireStringRegexCache;
-
-@property(nonatomic, strong) NSLock *lockPatternCache;
-@property(nonatomic, strong) NSMutableDictionary *regexPatternCache;
+@property(nonatomic, strong) NSCache *entireStringRegexCache;
```

Comment:
Since we're updating this anyway, we might as well add lightweight generics while we're here. Thoughts?

Reply:
There's something that also bugs me about declaring a property just to access the underlying ivar directly. I get that we're inheriting this code from elsewhere so no need to change it

Reply:
Yeah, I figure this should stay close to the original style. I can add the generics, though.